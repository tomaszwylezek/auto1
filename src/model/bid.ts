export interface IBid {
  id: string,
  carTitle: string,
  amount: number,
  created: string
}
