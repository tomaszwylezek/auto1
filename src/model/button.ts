export enum ButtonTheme {
  primary = 'primary',
  secondary = 'secondary',
  success = 'success',
  danger = 'danger',
  warning = 'warning',
  info = 'info',
  light = 'light',
  dark = 'dark',
  default = 'default',
  outlinePrimary = 'outline-primary',
  outlineSecondary = 'outline-secondary',
  outlineSuccess = 'outline-success',
  outlineDanger = 'outline-danger',
  outlineWarning = 'outline-warning',
  outlineInfo = 'outline-info',
  outlineLight = 'outline-light',
  outlineDark = 'outline-dark'
}

export enum Size {
  sm = 'sm',
  lg = 'lg'
}
