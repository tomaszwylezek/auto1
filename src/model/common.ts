import { Action } from 'redux';

export interface IAction<T> extends Action {
  payload: T;
}
