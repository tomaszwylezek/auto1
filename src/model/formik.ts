import React from 'react';

export interface FormikField {
  onChange: (e: React.ChangeEvent<any>) => void;
  onBlur: (e: any) => void;
  value: any;
  name: string;
}
