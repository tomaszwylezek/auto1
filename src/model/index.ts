export * from './store';
export * from './merchant';
export * from './bid';
export * from './formik';
