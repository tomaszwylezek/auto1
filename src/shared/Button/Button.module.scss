.button {
  &:hover {
    cursor: pointer;
  }

  &:disabled,
  &[disabled] {
    cursor: not-allowed;
  }
}

.link {
  &:hover {
    text-decoration: none;
  }
}
