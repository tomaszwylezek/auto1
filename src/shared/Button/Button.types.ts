import { ReactNode } from 'react';
import { ButtonTheme, Size } from 'model';

export interface IButtonOwnProps {
  to?: string;
  type?: 'button' | 'submit' | 'reset';
  theme?: ButtonTheme;
  className?: string;
  size?: Size;
  disabled?: boolean;
  onClick?: () => any;
  children?: ReactNode;
}

export interface IButtonProps extends IButtonOwnProps {}
