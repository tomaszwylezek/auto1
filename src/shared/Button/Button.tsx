import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';

// Models
import { ButtonTheme } from 'model';
import { IButtonProps } from './Button.types';

import styles from './Button.module.scss';

export const Button: React.FC<IButtonProps> = ({
  to,
  theme = ButtonTheme.primary,
  className,
  type = 'button',
  size,
  children,
  disabled,
  onClick
}: IButtonProps): JSX.Element => {
  const buttonClassNames: string = classNames('btn', `btn-${theme}`, styles.button, styles[theme], className, {
    [`btn-${size}`]: !!size
  });

  if (to) {
    return (
      <Link to={to} className={buttonClassNames}>
        {children}
      </Link>
    );
  }

  return (
    <button type={type} onClick={!disabled ? onClick : undefined} disabled={disabled} className={buttonClassNames}>
      {children}
    </button>
  );
};
