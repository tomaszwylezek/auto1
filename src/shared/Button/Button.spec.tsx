import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Button } from './Button';

// Models
import { ButtonTheme } from 'model';
import { IButtonProps } from './Button.types';

describe('Button component should', (): void => {
  it('renders basic component', (): void => {
    const component: ShallowWrapper<IButtonProps> = shallow(<Button>Anything</Button>);

    expect(component).toMatchSnapshot();
  });

  it('renders Link', (): void => {
    const component: ShallowWrapper<IButtonProps> = shallow(<Button to={'/foo'}>Anything</Button>);

    expect(component).toMatchSnapshot();
  });

  it('renders other theme', (): void => {
    const component: ShallowWrapper<IButtonProps> = shallow(
      <Button theme={ButtonTheme.outlineDanger}>Anything</Button>
    );

    expect(component.find('button').prop('className')).toContain('outline-danger');
  });
});
