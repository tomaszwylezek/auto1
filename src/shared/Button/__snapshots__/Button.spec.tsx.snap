// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should renders Link 1`] = `
<Link
  className="btn btn-primary button primary"
  to="/foo"
>
  Anything
</Link>
`;

exports[`Button component should renders basic component 1`] = `
<button
  className="btn btn-primary button primary"
  type="button"
>
  Anything
</button>
`;
